### PR TITLE
fix(docs): schema for uploaded config

### PR DESCRIPTION
**Pull request by fengelniederhammer**
## PR Checklist
- ~~[ ] All necessary documentation has been adapted.~~
- ~~[ ] The implemented feature is covered by an appropriate end-to-end test.~~

### CODE REVIEW

File: lapis2-docs/src/components/ConfigGenerator/configContext.tsx [hunk 1]
```diff
@@ -24,49 +24,37 @@ export type MetadataType = z.infer<typeof metadataTypeSchema>;
...
-    instanceName: string;
-    opennessLevel: OpennessLevel;
-    metadata: Metadata[];
-    primaryKey: string;
-    dateToSortBy: string;
-    partitionBy: string;
-    tableColumns: string[];
-    features: Feature[];
-};
+export type Schema = z.infer<typeof schemaSchema>;
```
Comment (JonasKellerer) [inline lapis2-docs/src/components/ConfigGenerator/configContext.tsx:51]: ;) schemaSchema

File: lapis2-docs/src/components/ConfigGenerator/MetadataWizard.tsx [hunk 2]
```diff
@@ -105,7 +105,7 @@ export function MetadataEditModal({ index, metadata }: { index: number; metadata
 
     const [metadataType, setMetadataType] = useState<MetadataType>(metadata.type);
     const [metadataName, setMetadataName] = useState(metadata.name);
-    const [generateIndex, setGenerateIndex] = useState(metadata.generateIndex);
+    const [generateIndex, setGenerateIndex] = useState(!!metadata.generateIndex);
```
Comment (JonasKellerer) [inline lapis2-docs/src/components/ConfigGenerator/MetadataWizard.tsx:108]: Shouldn't we also change the type of `generateIndex`? Or do we want to have explicitly in the config `generateIndex: false`. Then we would need to adapt this here and change the `handleSetMetadataType` accordingly

**Review by JonasKellerer — APPROVED** [review]
LGTM

File: lapis2-docs/src/components/ConfigGenerator/configContext.tsx (hunk 1 above)
Comment (fengelniederhammer) [reply lapis2-docs/src/components/ConfigGenerator/configContext.tsx:51]: Well, the schema in this file is: call the zod schema `somethingSchema` and the type `Something`. If you have a better suggestion than `schemaSchema`, I'm happy to hear it :D

Comment (JonasKellerer) [reply lapis2-docs/src/components/ConfigGenerator/configContext.tsx:51]: I have no better idea, but funny nonetheless. ;)

File: lapis2-docs/src/components/ConfigGenerator/MetadataWizard.tsx (hunk 2 above)
Comment (fengelniederhammer) [reply lapis2-docs/src/components/ConfigGenerator/MetadataWizard.tsx:108]: I checked in SILO and the code looks like it doesn't care whether `generateIndex` is present or not. So I would go for this simple implementation here (which always adds `generateIndex`)

**Pull request merged**